### PR TITLE
Update the logic for retraining an MNIST model

### PR DESCRIPTION
**Pull request by saudet**
I've hacked the `trainModel()` method to accept a custom image. When passed such an image, it retrains the model until it gets that image right. To make sure we don't overfit too much, it actually retrains using the whole MNIST dataset as well. I don't know how well that works in practice, but it makes for a good demo I think. The output of the retraining happening in the `runTests()` method looks like this:
```
INFO: *****TRAIN MODEL********
[main] INFO org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 1000 is 0.3224079661587816
[main] INFO org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 1100 is 0.15979046270622885
[main] INFO org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 1200 is 0.28417009034767055
[main] INFO org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 1300 is 0.3423950959362791
[main] INFO org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 1400 is 0.2561131170041967
custom feature 6 predicted as 6

INFO: *****TRAIN MODEL********
[main] INFO org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 1500 is 0.16176872633877773
[main] INFO org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 1600 is 0.41596402990994447
[main] INFO org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 1700 is 0.24138384336280194
[main] INFO org.deeplearning4j.optimize.listeners.ScoreIterationListener - Score at iteration 1800 is 0.3043453925919694
custom feature 6 predicted as 6
 prediction = 6
 prediction = 6
```

It's not exactly the gradient, but it's basically a form of _Asynchronous Stochastic Gradient Descent_ (ASGD):
https://blog.skymind.ai/distributed-deep-learning-part-1-an-introduction-to-distributed-training-of-neural-networks/

It also trains the initial model with "inverted" colors (black on white), which makes more sense if we expect all future images to be like that anyway.

Finally, to integrate that in the demo app and send the update to the server, the easiest way to de/serialize an `NDArray` is with `Nd4jBase64.base64String()/fromBase64()`.

Let me know you need anything else to get this working!

### CODE REVIEW

**Comment by johanvos** [comment]
looks great, thanks!
